src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -1009,4 +1014,69 @@ pub mod tests {
         let require_old_value_none = true;
         old_value_random(key, require_old_value_none, tests_require_old_value_none);
     }
+
+    #[test]
+    fn test_acquire_pessimistic_lock_should_not_exist() {
+        let engine = TestEngineBuilder::new().build().unwrap();
+
+        let (key, value) = (b"k", b"val");
+
+        // T1: start_ts = 3, commit_ts = 5, put key:value
+        must_succeed(&engine, key, key, 3, 3);
+        must_pessimistic_prewrite_put(&engine, key, value, key, 3, 3, true);
+        must_commit(&engine, key, 3, 5);
+
+        // T2: start_ts = 15, acquire pessimistic lock on k, with should_not_exist flag set.
+        let snapshot = engine.snapshot(Default::default()).unwrap();
+        let min_commit_ts = TimeStamp::zero();
+        let cm = ConcurrencyManager::new(min_commit_ts);
+        let start_ts = TimeStamp::new(15);
+        let for_update_ts = TimeStamp::new(15);
+        let need_old_value = true;
+        let need_value = false;
+        let mut txn = MvccTxn::new(start_ts, cm.clone());
+        let mut reader = SnapshotReader::new(start_ts, snapshot, true);
+        let _res = acquire_pessimistic_lock(
+            &mut txn,
+            &mut reader,
+            Key::from_raw(key),
+            key,
+            true,
+            0,
+            for_update_ts,
+            need_value,
+            min_commit_ts,
+            need_old_value,
+        )
+        .unwrap_err();
+
+        assert_eq!(cm.max_ts().into_inner(), 15);
+
+        // T3: start_ts = 8, commit_ts = max_ts + 1 = 16, prewrite a DELETE operation on k
```

Comment:
Emm, you're simulating manually committing with async commit 😂 not a big problem, though
Actually, personally I think only checking the current `max_ts` is enough...

Reply:
Got it.